pypesto/problem.py

```diff
@@ -232,24 +235,18 @@ def unfix_parameters(
         if not isinstance(parameter_indices, list):
             parameter_indices = [parameter_indices]
 
-        # first clean to be freed indices
-        for i_index, i_parameter in enumerate(parameter_indices):
+        # first clean to-be-freed indices
+        for i_parameter in parameter_indices:
             if i_parameter in self.x_fixed_indices:
+                i_index = self.x_fixed_indices.index(i_parameter)
                 self.x_fixed_indices.pop(i_index)
                 self.x_fixed_vals.pop(i_index)
 
         self.dim = self.dim_full - len(self.x_fixed_indices)
```

Comment:
duplicate of code in fix_parameters, move to seperate method?

Reply:
Potentially: move to `self.normalize()` ?

Reply:
both dim and x_free_indices can be properties too. comp. overhead for x_free_indices exists but should be small (O(n_par)).